webviz_subsurface/plugins/_pvt_plot.py

```diff
@@ -0,0 +1,640 @@
+                Input(self.uuid("ensemble"), "value"),
+                Input(self.uuid("pvtnum"), "value"),
+            ],
+        )
+        def _update_graph(
+            color_by: str, phase: str, ensembles: List[str], pvtnums: List[str]
+        ):
+            if not phase:
+                raise PreventUpdate
+            if ensembles is None:
+                ensembles = []
+            if not isinstance(ensembles, list):
+                ensembles = [ensembles]
+            if not isinstance(pvtnums, list):
+                pvtnums = [pvtnums]
+
+            data_frame = filter_data_frame(self.pvt_data_frame, ensembles, pvtnums)
+
+            if color_by == "ENSEMBLE":
+                colors = self.ensemble_colors
+            elif color_by == "PVTNUM":
+                colors = self.pvtnum_colors
+
+            layout = plot_layout(phase, color_by, self.plotly_theme["layout"])
+
+            traces = add_realization_traces(data_frame, color_by, colors, phase)
+
+            return {"data": traces, "layout": layout}
+
+        @app.callback(
+            [
+                Output(self.uuid("ensemble"), "multi"),
+                Output(self.uuid("ensemble"), "value"),
+            ],
+            [Input(self.uuid("color_by"), "value")],
+            [State(self.uuid("stored_ensemble"), "data")],
+        )
+        def _set_ensemble_selector(
+            color_by: str, stored_ensemble: pd.DataFrame
+        ) -> (bool, str):
+            """If ensemble is selected as color by, set the ensemble
+            selector to allow multiple selections, else use stored_ensemble
+            """
+
+            if color_by == "ENSEMBLE":
+                return True, self.ensembles
+
+            return (
+                False,
+                stored_ensemble.get("ENSEMBLE", self.ensembles[0]),
+            )
+
+        @app.callback(
+            [
+                Output(self.uuid("pvtnum"), "multi"),
+                Output(self.uuid("pvtnum"), "value"),
+            ],
+            [Input(self.uuid("color_by"), "value")],
+            [State(self.uuid("stored_pvtnum"), "data")],
+        )
+        def _set_pvtnum_selector(
+            color_by: str, stored_pvtnum: pd.DataFrame
+        ) -> (bool, str):
+            """If pvtnum is selected as color by, set the pvtnum
+            selector to allow multiple selections, else use stored_satnum
+            """
+
+            if color_by == "PVTNUM":
+                return True, self.pvtnums
+
+            return (
+                False,
+                stored_pvtnum.get("PVTNUM", self.pvtnums[0]),
+            )
+
+    def add_webvizstore(self):
+        return [
+            (
+                load_pvt_dataframe,
+                [
+                    {
+                        "ensemble_paths": self.ensemble_paths,
+                        "ensemble_set_name": "EnsembleSet",
+                    }
+                ],
+            )
+        ]
```

Comment:
This only supports the `ecl2df` scenario. Need to add some logic to generalize it. The suggestion assumes that you add something like `self.read_from_init_file = read_from_init_file` in `PvtPlot`'s `init`
```suggestion
    def add_webvizstore(self):
        return (
            [
                (
                    load_pvt_dataframe,
                    [
                        {
                            "ensemble_paths": self.ensemble_paths,
                            "use_init_file": self.read_from_init_file,
                        }
                    ],
                )
            ]
            if self.pvt_relative_file_path is None
            else [
                (
                    load_csv,
                    [
                        {
                            "ensemble_paths": self.ensemble_paths,
                            "csv_file": self.pvt_relative_file_path,
                        }
                    ],
                )
            ]
        )
```

Reply:
Good catch, that is definitely an important fix.